ext/standard/string.c

```diff
@@ -4530,6 +4531,11 @@ PHP_FUNCTION(parse_str)
 		sapi_module.treat_data(PARSE_STRING, res, &ret);
 		ZVAL_COPY_VALUE(arrayArg, &ret);
 	}
+
+	/* parse_str() tries to parse as much as possible (per treat_data behavior), hence we just ignore possible HashCollisionErrors */
+	if (EG(exception) && EG(exception)->ce == zend_ce_hash_collision_error) {
+		zend_clear_exception();
+	}
```

Comment:
Maybe. I'd also like to know what currently happens with this exception if it occurs during request parsing (i.e. max_input_vars > collision limit)